packages/tutanota-utils/lib/Utils.ts

```diff
@@ -511,3 +510,28 @@ export function mapObject<K extends string | number | symbol, V, R>(mapper: (arg
 
 	return newObj
 }
+
+/**
+ * Run jobs with defined max parallelism.
+ */
+export class BoundedExecutor {
+	private runningJobsCount: number = 0
+	private currentJob: Promise<unknown> = Promise.resolve()
+
+	constructor(private readonly maxParallelJobs: number) {}
+
+	async run<T>(job: () => Promise<T>): Promise<T> {
+		while (this.runningJobsCount === this.maxParallelJobs) {
+			await this.currentJob
+		}
+		this.runningJobsCount++
+
+		try {
+			const jobResult = job()
+			this.currentJob = jobResult.catch(noOp)
```

Comment:
I think it is worth mentioning in the docs that this will eat all errors job is producing

also, is the type right? looks like the type of run() is actually 
```
run(job: () => Promise<T>) => Promise<T | undefined>
```
because the noOp does not return a value?

Reply:
it will give you back a promise that might be rejected and you should catch it. I think that's enough.

`noop` is only used for the `this.currentJob`, we do not return that